### PR TITLE
Use arp command to get switch mac address

### PR DESCRIPTION
**Pull request by cxhong**
The switchdiscover command couldn't find Mellox switch's Mac address via snmpwalk,  we are change it to use arp -n .  this is for issue #1036

### CODE REVIEW

**Comment by cxhong** [comment]
@daniceexi  can you review this changes?

File: perl-xCAT/xCAT/Utils.pm [hunk 1]
```diff
@@ -4711,5 +4711,37 @@ sub get_nmapversion {
     return $nmap_version;
 }
 
+#--------------------------------------------------------------------------------
```
Comment (daniceexi) [inline perl-xCAT/xCAT/Utils.pm]: @cxhong There's already a subroutine fro get mac from arp

~~~~
./xCAT-server/lib/perl/xCAT/SvrUtils.pm:=head3  get_mac_by_arp
~~~~

**Comment by cxhong** [comment]
@daniceexi 
the subroutine get_mac_by_arp() in the SvrUtils.pm will call pping first
````
 my @output = xCAT::Utils->runcmd("/opt/xcat/bin/pping $noderange", -1);
``````
and it will failed because the switch is not defined in the xcat database yet and pping subroutine will return output as "unreachable"

So, add a simple get_macbyarp subroutine in the Utils.pm works better for this case.

**Comment by daniceexi** [comment]
@cxhong it's not a good idea to add more than one subroutine to do similar thing but in different utils files. Especially that user cannot figure them out easier that which one should be used through the ``description`` for the subroutines.

I suggest to add a new argument 'noping' for ``SvrUtils:: get_mac_by_arp`` to indicate that don't run the ``pping`` against the node or ip, but just run arp to get the mac.

**Comment by cxhong** [comment]
@daniceexi , I modified the code with:
1) used Svrutils::get_mac_by_arp subroutine
2) add message if snmp port is disabled.

File: xCAT-server/lib/xcat/plugins/switchdiscover.pm [hunk 2]
```diff
@@ -855,19 +860,35 @@ sub snmp_scan {
...
+        }
+
         my $vendor = get_snmpvendorinfo($request, $ip);
         if ($vendor) {
-            my $mac = xCAT::Utils->get_macbyarp($ip);
+            my $display = "";
+            my $nopping = "nopping";
+            my $output = xCAT::SvrUtils->get_mac_by_arp([$ip], $display, $nopping);
+            my $mac;
+            foreach my $node (keys %{$output}) {
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/switchdiscover.pm]: if $output is a hash, you can use ``$output->{$ip}`` to get the value directly instead of a loop.

Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/switchdiscover.pm]: The indent should be 4 spaces.

File: xCAT-server/lib/xcat/plugins/switchdiscover.pm [hunk 3]
```diff
@@ -826,13 +827,17 @@ sub snmp_scan {
     my $ranges = get_ip_ranges($request);
 
 
-    #use nmap to find if snmp port is enabled  
-    # only open port will be scan
+    ##########################################################
+    #use nmap to parse the ip range and possible output from the command:
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/switchdiscover.pm]: Do we need to add the output example for nmap 5.10 and earlier?

**Comment by cxhong** [comment]
@daniceexi  Thanks,  code modified.

**Comment by cxhong** [comment]
@daniceexi , can you review this again?  Thanks

**Comment by daniceexi** [comment]
I have reviewed the change and agree to merge. Thanks. @cxhong

**Pull request merged**